src/main/java/io/jenkins/plugins/audit/listeners/BuildListener.java

```diff
@@ -48,9 +49,41 @@ public void onStarted(Run run, TaskListener listener) {
     }
 
     /**
-     * Returns a registered {@link BuildStartListener} instance.
+     * Fired when a build is completed, event logged via Log4j-audit.
+     *
+     * @param run of type Run having the build information
+     * @param listener of type TaskListener that the onCompleted method expects
+     */
+    @Override
+    public void onCompleted(Run run, TaskListener listener) {
+        BuildFinish buildFinish = LogEventFactory.getEvent(BuildFinish.class);
+
+        List causeObjects = run.getCauses();
+        List<String> causes = new ArrayList<>(causeObjects.size());
+        for (Object cause: causeObjects) {
+            Cause c = (Cause)cause;
+            causes.add(c.getShortDescription());
+        }
+
+        buildFinish.setBuildNumber(run.getNumber());
+        buildFinish.setCause(causes);
+        buildFinish.setProjectName(run.getParent().getFullName());
+        buildFinish.setTimestamp(new Date(run.getStartTimeInMillis()).toString());
```

Comment:
We want the finish time here, not the start.

Reply:
Ah!, Sorry about that. Thanks for pointing out. :)